### DISCUSSION TITLE
Error creating VLAN

### DISCUSSION THREAD

**Discussion by miloserdoff** [apache/cloudstack#10190] [Q&A] [answered]
##### ISSUE TYPE

 * Other

##### COMPONENT NAME

~~~
VLAN
~~~

##### CLOUDSTACK VERSION

~~~
4.18.2.4
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~
When creating a VM from a disk image, Cloudstack cannot create a VLAN

2024-11-08 06:44:20,824 WARN [resource.wrapper.LibvirtStartCommandWrapper] (agentRequest-Handler-5:null) (logid:d046b384) InternalErrorException
com.cloud.exception.InternalErrorException: Failed to create vnet 803
~~~

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by DaanHoogland** [comment]
@miloserdoff can you add some more information, please?

What ui-function or API are you using? Did you convert a disk image to a template or upload a template, etc?

Do you see just this WARN message or are there more relevant log messages?
.
thanks,

**Comment by weizhouapache** [comment]
@miloserdoff 
can you share the full agent.log around it ?

**Comment by miloserdoff** [comment]
@weizhouapache @DaanHoogland 

Thanks for the reply. I'm using a pre-built CentOS image, but it's not the image itself.

My Cloudstack Agent log file is throwing an error, as I understand it, when trying to create a virtual router.

2024-11-08 08:34:24,675 WARN [resource.wrapper.LibvirtStartCommandWrapper] (agentRequest-Handler-1:null) (logid:cbb2042c) InternalErrorException
com.cloud.exception.InternalErrorException: Failed to create vnet 761: Error: argument "enp129s0f1np1.761" is wrong: "name" not a valid ifname/usr/share/cloudstack-common/scripts/vm/network/vnet/modifyvlan.sh: line 37: /proc/sys/net/ipv6
/conf/enp129s0f1np1.761/disable_ipv6: No such file or directoryCannot find device "enp129s0f1np1.761"Failed to create vlan 761 on pif: enp129s0f1np1.
 at com.cloud.hypervisor.kvm.resource.BridgeVifDriver.createVnet(BridgeVifDriver.java:320)
 at com.cloud.hypervisor.kvm.resource.BridgeVifDriver.createVnetBr(BridgeVifDriver.java:302)
 at com.cloud.hypervisor.kvm.resource.BridgeVifDriver.plug(BridgeVifDriver.java:222)
 at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.createVif(LibvirtComputingResource.java:3277)
 at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.createVifs(LibvirtComputingResource.java:2891)
 at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtStartCommandWrapper.execute(LibvirtStartCommandWrapper.java:84)
 at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtStartCommandWrapper.execute(LibvirtStartCommandWrapper.java:49)
 at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtRequestWrapper.execute(LibvirtRequestWrapper.java:78)
 at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.executeRequest(LibvirtComputingResource.java:1869)
 at com.cloud.agent.Agent.processRequest(Agent.java:663)
 at com.cloud.agent.Agent$AgentRequestHandler.doTask(Agent.java:1086)
 at com.cloud.utils.nio.Task.call(Task.java:83)
 at com.cloud.utils.nio.Task.call(Task.java:29)
 at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 at java.base/java.lang.Thread.run(Thread.java:829)

I also attach the output of the "ip a" command to demonstrate the network configuration

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
inet 127.0.0.1/8 scope host lo
valid_lft forever preferred_lft forever
inet6 ::1/128 scope host
valid_lft forever preferred_lft forever
2: enxbe3af2b6059f: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
link/ether be:3a:f2:b6:05:9f brd ff:ff:ff:ff:ff:ff
3: enp129s0f0np0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
 link/ether 7c:c2:55:32:00:90 brd ff:ff:ff:ff:ff:ff
4: enp129s0f1np1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master cloudbr0 state UP group default qlen 1000
 link/ether 7c:c2:55:32:00:91 brd ff:ff:ff:ff:ff:ff
5: cloudbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
 link/ether 42:c1:eb:1f:33:33 brd ff:ff:ff:ff:ff:ff
 inet 192.168.127.57/24 brd 192.168.127.255 scope global cloudbr0
 valid_lft forever preferred_lft forever
 inet6 fe80::40c1:ebff:fe1f:3333/64 scope link
 valid_lft forever preferred_lft forever
6: cloud0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
 link/ether 82:8d:f9:c3:b0:e0 brd ff:ff:ff:ff:ff:ff
 inet 169.254.0.1/16 scope global cloud0
 valid_lft forever preferred_lft forever
 inet6 fe80::808d:f9ff:fec3:b0e0/64 scope link
 valid_lft forever preferred_lft forever
8: vnet0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
 link/ether fe:00:a9:fe:ec:24 brd ff:ff:ff:ff:ff:ff
 inet6 fe80::fc00:a9ff:fefe:ec24/64 scope link
 valid_lft forever preferred_lft forever
10: vnet3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
 link/ether fe:00:91:00:00:06 brd ff:ff:ff:ff:ff:ff
 inet6 fe80::fc00:91ff:fe00:6/64 scope link
 valid_lft forever preferred_lft forever
11: vnet5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloudbr0 state UNKNOWN group default qlen 1000
 link/ether fe:00:b6:00:00:0c brd ff:ff:ff:ff:ff:ff
 inet6 fe80::fc00:b6ff:fe00:c/64 scope link
 valid_lft forever preferred_lft forever
13: vnet6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master cloud0 state UNKNOWN group default qlen 1000
 link/ether fe:00:a9:fe:3f:52 brd ff:ff:ff:ff:ff:ff
 inet6 fe80::fc00:a9ff:fefe:3f52/64 scope link
 valid_lft forever preferred_lft forever

**Comment by rajujith** [comment]
@miloserdoff Could you try changing the interface names to enp1, enx0 ?

**Comment by miloserdoff** [comment]
@rajujith Hello. I changed the interface names and decided to look at the Cloudstack Agent log. It gave me an error:

com.cloud.exception.InternalErrorException: Failed to create vnet 805: Cannot find device "br-805" /usr/share/cloudstack-common/scripts/vm/network/vnet/modifyvlan.sh: line 37: /proc/sys/net/ipv6/conf/br-805.805/disable_ipv6: No such file or directory Cannot find device "br-805.805" Failed to create vlan 805 on pif: br-805.

**Comment by rajujith** [comment]
@miloserdoff can you check for any stale bridge configurations on the host and fix them? 
Below is an example that works properly : 

`/usr/share/cloudstack-common/scripts/vm/network/vnet/modifyvlan.sh -v 1556 -p eth1 -b breth1-1556 -o add`

the -p should be passed with the physical interface name but in your log, its having a bridge interface. 

`"Usage: %s: -o <op>(add | delete) -v <vlan id> -p <pif> -b <bridge name> -d <delete bridge>(true | false)\n"`

**Comment by weizhouapache** [comment]
> @miloserdoff can you check for any stale bridge configurations on the host and fix them? Below is an example that works properly :
> 
> `/usr/share/cloudstack-common/scripts/vm/network/vnet/modifyvlan.sh -v 1556 -p eth1 -b breth1-1556 -o add`
> 
> the -p should be passed with the physical interface name but in your log, its having a bridge interface.
> 
> `"Usage: %s: -o <op>(add | delete) -v <vlan id> -p <pif> -b <bridge name> -d <delete bridge>(true | false)\n"`

maybe there is no physical nic associated to the bridge ?

**Comment by bl0way** [comment, marked as answer]
Hello,

I got the same issue (fresh  `4.20.0` cloudstack installation).
This comes from the name of the interface that is too long. Linux interfaces should be **15** characters maximum. That is why the ip command is failing in the cloudstack script. 
In this case `enp129s0f1np1.761` is 17 characters.

I fixed it by renaming interfaces.

**Comment by aki-luxor** [comment]
> Hello,
> 
> I got the same issue (fresh `4.20.0` cloudstack installation). This comes from the name of the interface that is too long. Linux interfaces should be **15** characters maximum. That is why the ip command is failing in the cloudstack script. In this case `enp129s0f1np1.761` is 17 characters.
> 
> I fixed it by renaming interfaces.

How did you change the interface name? for me after changing interface name its getting new error. Ubuntu 22.04. 

` Pool-196:[ctx-57c05f18]) (logid:5feaecac) Failed to handle host connection: com.cloud.exception.ConnectionException: Incorrect Network setup on agent, Reinitialize agent after network names are setup, details : Can not find network: br1
`

**Comment by rajujith** [comment]
@aki-luxor here are some places you can cross-check.
- agent.properties file on the KVM host containing information on the bridges used for the cloudstack zone. path: /etc/cloudstack/agent/agent.properties
- The current bridge configuration on the KVM host.
- netplan configuration
- CloudStack zone traffic labels for KVM

**Reply by aki-luxor** [reply]
> @aki-luxor here are some places you can cross-check.
> 
> * agent.properties file on the KVM host containing information on the bridges used for the cloudstack zone. path: /etc/cloudstack/agent/agent.properties
> * The current bridge configuration on the KVM host.
> * netplan configuration
> * CloudStack zone traffic labels for KVM

Hi @rajujith, it looks like name change is working but now I am getting another error on the KVM host. Screenshot attached it for the kvm host trying to run systemvm. 

<img width="824" alt="image" src="https://github.com/user-attachments/assets/ddb8ac1a-cc91-42e5-addc-70a09ab1757d" />

Everything works fine if I use a interface that has shorter name by default, so I am assuming this is all linked to interface name but not sure how.

**Reply by rajujith** [reply]
@aki-luxor  when does it throw this error? for any operation on an instance or network? Can you try restarting the network with clean up.

**Reply by aki-luxor** [reply]
> @aki-luxor when does it throw this error? for any operation on an instance or network? Can you try restarting the network with clean up.

Just did a fresh install of OS and cloudstack and same error.

**Reply by weizhouapache** [reply]
I saw the same error messages with `patched.sh` in my SSVM.
When the execution failed, ACS run the script again until it succeeded.
I think we can ignore the error

**Reply by aki-luxor** [reply]
@weizhouapache I agree it goes away after a while but still connection doesnt work as expected. 

@rajujith  I did complete fresh install after installing clean os, same result. 


<img width="1046" alt="Screenshot 2025-01-28 at 6 16 38 PM" src="https://github.com/user-attachments/assets/4386237e-fb3b-4dc7-ae2f-77cfcb248b2f" />
